### PR TITLE
Some crash and small bug fixes

### PR DESCRIPTION
**Pull request by btepu002xamk**

### CODE REVIEW

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 1]
```diff
@@ -78,6 +78,13 @@ const MetaformEditorRightDrawerFeature: FC<Props> = ({
     updateSelected();
   }, [ sectionIndex, fieldIndex, pendingForm ]);
 
+  useEffect(() => {
+    setSelectMemberGroupEnabled(false);
+    setMemberGroupOptIndex(undefined);
+    setSelectedMemberGroupId(undefined);
+    setSelectedMemberGroupPermission(NOT_SELECTED);
+  }, [ sectionIndex, fieldIndex ]);
+
```
Comment (Chi-Chu319) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:87]: Why is this necessary? Shouldn't checkIfMemberGroupsAreSelected handle the change of selected field already?
```
   /**
   * Empties member group settings
   */
  const emptyMemberGroupSettings = () => {
    setSelectMemberGroupEnabled(false);
    setMemberGroupOptIndex(undefined);
    setSelectedMemberGroupId(undefined);
    setSelectedMemberGroupPermission(NOT_SELECTED);
  };

  /**
   * Set member groups for selected field to the found first group
   */
  const checkIfMemberGroupsAreSelected = () => {
    if (sectionIndex === undefined || fieldIndex === undefined) {
      emptyMemberGroupSettings();
      return;
    }

    const updatedSelectedField = pendingForm.sections?.[sectionIndex].fields?.[fieldIndex];

    if (!updatedSelectedField || !MetaformUtils.fieldTypesAllowVisibility.includes(updatedSelectedField.type)) {
      emptyMemberGroupSettings();
      return;
    }

    const foundOptionIndex = updatedSelectedField.options?.findIndex(option => MetaformUtils.getOptionPermissionGroup(option) !== undefined);

    if (foundOptionIndex !== -1 && foundOptionIndex !== undefined) {
      const [ groupId, permission ] = MetaformUtils.getOptionPermissionGroup(updatedSelectedField.options![foundOptionIndex])!;

      setSelectMemberGroupEnabled(true);
      setSelectedMemberGroupId(groupId);
      setSelectedMemberGroupPermission(permission !== MemberGroupPermission.NOTIFY ? permission : NOT_SELECTED);
      setMemberGroupOptIndex(foundOptionIndex);
    } else {
      emptyMemberGroupSettings();
    }
  };

  useEffect(() => {
    updateSelected();
  }, [ sectionIndex, fieldIndex, pendingForm ]);

  useEffect(() => {
    checkIfMemberGroupsAreSelected();
  }, [ sectionIndex, fieldIndex, selectedField?.options?.length ]);
```
Maybe this one is better so you dont need an extra useeffect hook. And you could possibly replace the part where sets all the member group settings empty in removeAllPermissionGroups with emptyMemberGroupSettings

**Comment by btepu002xamk** [comment]
-Fixed crashing when changing from field where was more options than current field and caused crash when option index was not found.
-Sometimes field member group textfield rendered wrong data.
-Also added option to non select member group.

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 2]
```diff
@@ -567,6 +570,9 @@ const MetaformEditorRightDrawerFeature: FC<Props> = ({
           value={ selectedMemberGroupId }
           onChange={ event => handleMemberGroupChange(event.target.value) }
         >
+          <MenuItem value={ NOT_SELECTED } key={uuid4()}>
+            { strings.draftEditorScreen.editor.memberGroups.noMemberGroupSelected }
+          </MenuItem>
```
Comment (Chi-Chu319) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:575]: What's the point of this, why would user want to select an empty group?

File: src/components/editor/metaform-editor-right-drawer-feature.tsx [hunk 3]
```diff
@@ -517,14 +523,12 @@ const MetaformEditorRightDrawerFeature: FC<Props> = ({
...
    * Render member group role options
    *
-   * @param field selected metaform field
    */
-  const renderMemberGroupPermissionSelect = (field: MetaformField) => {
-    if (memberGroupOptIndex === undefined || selectedMemberGroupId === undefined) {
+  const renderMemberGroupPermissionSelect = () => {
+    if (memberGroupOptIndex === undefined || selectedMemberGroupId === undefined || selectedMemberGroupId === NOT_SELECTED || !selectedField) {
       return null;
     }
```
Comment (Chi-Chu319) [inline src/components/editor/metaform-editor-right-drawer-feature.tsx:530]: Why using the selected field instead of the field from function prop? the field is just the selected field after null check

**Review by Chi-Chu319 — CHANGES_REQUESTED** [review]

File: src/components/editor/metaform-editor-right-drawer-feature.tsx (hunk 2 above)
Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:575]: Used to delete member groups from only 1 option.
Currently when u switch member group switch off, it deletes member groups from all options.

Comment (Chi-Chu319) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:575]: Users can just select no permission from the permission select them the permission for that group is disabled

Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:575]: We though its little more clearly when there is not selected option also in member groups.

File: src/components/editor/metaform-editor-right-drawer-feature.tsx (hunk 3 above)
Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx]: Fixed that back

File: src/components/editor/metaform-editor-right-drawer-feature.tsx (hunk 1 above)
Comment (btepu002xamk) [reply src/components/editor/metaform-editor-right-drawer-feature.tsx:87]: That suggestion is not working for some reason. Renders wrong values sometimes.

**Review by Chi-Chu319 — APPROVED** [review]
